editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/RestConfigEditor.java

```diff
@@ -194,9 +194,24 @@ private String getTextForImage(String text) {
 		if (compareTextAndTag(text, RestConfigConstants.POST_VERB) != null) {
 			return RestConfigConstants.POST_VERB;
 		}
+		if (compareTextAndTag(text, RestConfigConstants.PATCH_VERB) != null) {
+			return RestConfigConstants.PATCH_VERB;
+		}
 		if (compareTextAndTag(text, RestConfigConstants.DELETE_VERB) != null) {
 			return RestConfigConstants.DELETE_VERB;
 		}
+		if (compareTextAndTag(text, RestConfigConstants.HEAD_VERB) != null) {
+			return RestConfigConstants.HEAD_VERB;
+		}
+		if (compareTextAndTag(text, RestConfigConstants.TRACE_VERB) != null) {
+			return RestConfigConstants.TRACE_VERB;
+		}
+		if (compareTextAndTag(text, RestConfigConstants.CONNECT_VERB) != null) {
+			return RestConfigConstants.CONNECT_VERB;
+		}
+		if (compareTextAndTag(text, RestConfigConstants.OPTIONS_VERB) != null) {
+			return RestConfigConstants.OPTIONS_VERB;
+		}
 		return null;
```

Comment:
if there was a NPE, I guess it was because null is returned here.
What about throwing an Exception?
Or handling the null return in a better way in calling methods?

Reply:
from where the list of supported verbs comes from? What if there is a new one? What if there is a typo in the document? Would be nice to be able to open the editor and just point that one verb is invalid instead of having a NPE.